core/src/main/java/com/klaytn/caver/wallet/keyring/KeyStore.java

```diff
@@ -0,0 +1,1160 @@
+package com.klaytn.caver.wallet.keyring;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.klaytn.caver.utils.Utils;
+import org.bouncycastle.crypto.digests.SHA256Digest;
+import org.bouncycastle.crypto.generators.PKCS5S2ParametersGenerator;
+import org.bouncycastle.crypto.generators.SCrypt;
+import org.bouncycastle.crypto.params.KeyParameter;
+import org.web3j.crypto.CipherException;
+import org.web3j.crypto.Hash;
+import org.web3j.utils.Numeric;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.IOException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Represents a KeyStore DTO(Data transfer Object) class according to KIP-3.
+ * For more details, please see below link.
+ * http://kips.klaytn.com/KIPs/kip-3
+ */
+public class KeyStore {
+    /**
+     * Keystore Format V4 version
+     */
+    public static final int KEY_VERSION_V3 = 3;
+
+    /**
+     * Keystore Format V3 version
+     */
+    public static final int KEY_VERSION_V4 = 4;
+
+    /**
+     * An address in KeyStore
+     */
+    private String address;
+
+    /**
+     * An crypto in KeyStore V3
+     */
+    private KeyStore.Crypto crypto;
+
+    /**
+     * An keyring in KeyStore V4
+     */
+    private List keyRing;
+
+    /**
+     * An ID in KeyStore
+     */
+    private String id;
+
+    /**
+     * A Version in KeyStore
+     */
+    private int version;
+
+    /**
+     * Creates a KeyStore instance.
+     */
+    public KeyStore() {
+    }
+
+    /**
+     * Setter function of keyRing.
+     * @param keyRing
+     */
+    @JsonSetter("keyring")
```

Comment:
The variable name is keyRing, but does the keystore JSON contain the key value as `keyring`?

Reply:
Yes, It contains KeyStoreV4 same as "crypto" key in KeyStoreV3